Plugins/Public/base_plugin/PlayerCommands.cpp

```diff
@@ -1882,6 +1882,151 @@ namespace PlayerCommands
+			{
+				auto& zone = cfield->zone;
+				if (!zone->lootableZone)
+					continue;
+
+				if (minMiningDistance > 0 && cfield->near_field(pos))
+				{
+					PrintUserCmdText(client, L"You can't deploy inside a mining field!");
+					return false;
+				}
+				// pretend zone is actually centered 0,0,0 and unrotated
+				// Subtract zone positon from player position so their relative position is the same
+				Vector playerPos = pos;
+				playerPos.x -= zone->vPos.x;
+				playerPos.y -= zone->vPos.y;
+				playerPos.z -= zone->vPos.z;
+				
+				//Then rotate player position by inverse of original zone position, to make the relative positions truly in sync
+
+				Matrix InvertedZoneRot = InverseMatrix(zone->mRot);
+				playerPos = VectorMatrixMultiply(playerPos, InvertedZoneRot);
+
+				//Now, player position is effectively a vector from center of mining zone to player position.
+				//We normalize the vector, then apply original zone rotation and stretch it by zone sizes
+				Vector V2 = NormalizeVector(playerPos);
+				Matrix EllipseSizeMat = { {{zone->vSize.x,0,0},{0,zone->vSize.y,0},{0,0,zone->vSize.z}} };
+				V2 = VectorMatrixMultiply(V2, EllipseSizeMat);
+				V2 = VectorMatrixMultiply(V2, zone->mRot);
+				
+				//Now that we have our aproximated closest point to the mining zone, we move it back so comparison with player position yields proper result
+				V2.x += zone->vPos.x;
+				V2.y += zone->vPos.y;
+				V2.z += zone->vPos.z;
+
+				float distance = HkDistance3D(V2, pos);
+
+				if (distance < minMiningDistance)
+				{
+					if (zone->idsName)
+					{
+						PrintUserCmdText(client, L"Distance to %ls too close, minimum distance: %um.", HkGetWStringFromIDS(zone->idsName).c_str(), static_cast<uint>(minMiningDistance));
+					}
+					else
+					{
+						const GoodInfo* gi = GoodList::find_by_id(zone->lootableZone->dynamic_loot_commodity);
+						PrintUserCmdText(client, L"Distance to unnamed %ls field too close, minimum distance: %um.", HkGetWStringFromIDS(gi->iIDSName).c_str(), static_cast<uint>(minMiningDistance));
+					}
+					return false;
+				}
+			}
+		}
+
+		// Solars
+		bool foundSystemMatch = false;
+		for (CSolar* solar = reinterpret_cast<CSolar*>(CObject::FindFirst(CObject::CSOLAR_OBJECT)); solar;
+			solar = reinterpret_cast<CSolar*>(CObject::FindNext()))
+		{
+			//solars are iterated on per system, we can stop once we're done scanning the last solar in the system we're looking for.
+			if (solar->iSystem != systemID)
+			{
+				if (foundSystemMatch)
+					break;
+				continue;
+			}
+			else
+			{
+				foundSystemMatch = true;
+			}
+
+			float distance = HkDistance3D(solar->get_position(), pos);
+			switch (solar->iType)
+			{
+				case OBJ_PLANET:
+				case OBJ_MOON:
+				{
+					if (distance < minPlanetDistance + solar->get_radius()) // In case of planets, we only care about distance from actual surface, since it can vary wildly

```

Comment:
```suggestion
					if (distance < (minPlanetDistance + solar->get_radius())) // In case of planets, we only care about distance from actual surface, since it can vary wildly
```